src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -234,8 +234,7 @@ public DateTime(int year, int month, int day, int hour, int minute, int second,
         //
         public DateTime(int year, int month, int day, int hour, int minute, int second, Calendar calendar)
         {
-            if (calendar == null)
-                throw new ArgumentNullException(nameof(calendar));
+            ArgumentNullException.ThrowIfNull(calendar, nameof(calendar));
```

Comment:
I'd rather not start changing such null checks.  The plan is to switch to using the `!!` syntax once it's available, which should be soon.

Are these argument validation changes necessary for the purpose of this PR?